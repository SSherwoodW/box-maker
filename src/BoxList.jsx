import { useState } from "react";
import { v4 as uuid } from "uuid";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";

function BoxList() {
    const [boxes, setBoxes] = useState([])

    const addBox = box => {
        let newBox = { ...box, id: uuid() }
        setBoxes(boxes => [...boxes, newBox])
        console.log(newBox)
    }
    console.log(boxes)

    const removeBox = (id) => {
        setBoxes(boxes => boxes.filter(box => box.id !== id))
    }

    const boxComponents = boxes.map(box => (
        <Box
            key={box.id}
            id={box.id}
            width={box.width}
            height={box.height}
            backgroundColor={box.backgroundColor}
            handleRemove={removeBox}
        />
    ))

    return (
        <div >
            <NewBoxForm classname="BoxForm" addBox={addBox} />
            {boxComponents}

        </div>
    )

}

export default BoxList;