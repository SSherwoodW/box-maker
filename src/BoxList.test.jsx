import { describe, it, expect } from 'vitest';
import { render, fireEvent, queryByText } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(getByLabelText, getByText, fireEvent) {
    // Simulate adding a new box
        fireEvent.change(getByLabelText("Background Color:"), { target: { value: "teal" } });
        fireEvent.change(getByLabelText("Width:"), { target: { value: "50px" } });
        fireEvent.change(getByLabelText("Height:"), { target: { value: "50px" } });
        fireEvent.click(getByText("Add Box"));
}
describe('BoxList', () => {
    it('renders without crashing', () => {
        render(<BoxList />)
    })

    it('matches snapshot', () => {
        const { asFragment } = render(<BoxList />)
        expect(asFragment).toMatchSnapshot()
    })

    it('can add a new box', () => {
        const { getByLabelText, getByText } = render(<BoxList />);

       addBox(getByLabelText, getByText, fireEvent);

        expect(getByText("X")).toBeTruthy();
    })

    it('can remove a box', () => {
        const { getByLabelText, getByText, queryByText } = render(<BoxList />)
        
        addBox(getByLabelText, getByText, fireEvent);

        expect(getByText("X")).toBeTruthy();

        fireEvent.click(getByText("X"));
        
        expect(queryByText("X")).toBeFalsy();

    })
})