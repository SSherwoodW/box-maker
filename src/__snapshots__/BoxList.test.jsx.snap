// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoxList > matches snapshot 1`] = `[Function]`;
